### PR TITLE
Remove -Wno-sign-compare

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by Nekotekina** [comment]
Scary?

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 1]
```diff
@@ -861,7 +861,7 @@ error_code sceNpTrophyUnlockTrophy(u32 context, u32 handle, s32 trophyId, vm::pt
 	const std::string& config_path = vfs::get("/dev_hdd0/home/" + Emu.GetUsr() + "/trophy/" + ctxt->trp_name + "/TROPCONF.SFM");
 	const u32 unlocked_platinum_id = ctxt->tropusr->GetUnlockedPlatinumID(trophyId, config_path);
 
-	if (unlocked_platinum_id != SCE_NP_TROPHY_INVALID_TROPHY_ID)
+	if (unlocked_platinum_id + 1)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: this is bullshit.
it completely removes readability

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 2]
```diff
@@ -885,7 +885,7 @@ error_code sceNpTrophyUnlockTrophy(u32 context, u32 handle, s32 trophyId, vm::pt
 		// Enqueue popup for the regular trophy
 		show_trophy_notification(ctxt, trophyId);
 
-		if (unlocked_platinum_id != SCE_NP_TROPHY_INVALID_TROPHY_ID)
+		if (unlocked_platinum_id + 1)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: same here

File: rpcs3/Emu/Cell/PPUAnalyser.cpp [hunk 3]
```diff
@@ -719,7 +719,7 @@ void ppu_module::analyse(u32 lib_toc, u32 entry)
 	// Clean TOCs
 	for (auto&& pair : fmap)
 	{
-		if (pair.second.toc == -1)
+		if (pair.second.toc + 1 == 0)
```
Comment (AniLeo) [inline rpcs3/Emu/Cell/PPUAnalyser.cpp]: weird syntax for no reason, older version is more clear

**Comment by CookiePLMonster** [comment]
If signed/unsigned compares are so bad, out of the two I'd rather prefer seeing comparisons against `u32(-1)` tbh.

**Pull request merged**